### PR TITLE
Chore: update to use ES6 classes (refs #7849)

### PR DESCRIPTION
**Pull request by qlaire**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ x ] Other, please explain:

**What changes did you make? (Give an overview)**
Update constructor functions to use ES6 classes. Also, changed one constructor function (rule-fixer.js) to a plain object since it does not have any internal state.

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: lib/config/autoconfig.js [hunk 1]
```diff
@@ -65,17 +65,16 @@ function makeRegistryItems(rulesConfig) {
...
-function Registry(rulesConfig) {
-    this.rules = (rulesConfig) ? makeRegistryItems(rulesConfig) : {};
-}
-
-Registry.prototype = {
+class Registry {
 
-    constructor: Registry,
+    /**
+     * @param   {rulesConfig} [rulesConfig] Hash of rule names and arrays of possible configurations
```
Comment (kaicataldo) [inline lib/config/autoconfig.js]: Nitpick: Can we get rid of the extra whitespace after `@param`?

File: lib/util/rule-fixer.js [hunk 2]
```diff
@@ -34,14 +34,9 @@ function insertTextAt(index, text) {
...
  * Creates code fixing commands for rules.
- * @constructor
  */
-function RuleFixer() {
-    Object.freeze(this);
-}
 
-RuleFixer.prototype = {
-    constructor: RuleFixer,
+const ruleFixer = {
```
Comment (gyandeeps) [inline lib/util/rule-fixer.js]: We want to freeze this object so that we are consistent with what we had previously.

**Comment by qlaire** [comment]
@gyandeeps @kaicataldo thanks so much for reviewing! I've made the two changes you suggested.

**Comment by kaicataldo** [comment]
Leaving this open a little bit longer so that we can get a few more pairs of eyes on this.

**Comment by ilyavolodin** [comment]
LGTM. Thanks for contributing!

**Pull request merged**